### PR TITLE
calling exp_on_negative_values() once

### LINKED ISSUE #88861: Fixed-point Softmax() calls exp_on_negative_values() twice
Fixed-point `Softmax()` calls `exp_on_negative_values()` twice per element: once to calculate the value of the exp of each element
https://github.com/tensorflow/tensorflow/blob/4266e9ab53bb3f7fca0dd816b965b82862b69b4e/tensorflow/lite/kernels/internal/reference/softmax.h#L131
and another one a few lines earlier to calculate the sum of all exps
https://github.com/tensorflow/tensorflow/blob/4266e9ab53bb3f7fca0dd816b965b82862b69b4e/tensorflow/lite/kernels/internal/reference/softmax.h#L109-L110
The `exp_on_negative_values()` function can be rather slow, taking most of the time of `Softmax()` (which can be especially concerning in embedded implementations), so calling it twice per element makes the whole `Softmax()` function be about twice as slow as necessary.

Could this be optimized so that it is only called once per element?  Store the exp results in a temporary array (for example, reusing `output_data`), computing the sum as they're stored, and then applying the scaling to the precomputed exp results rather than computing them again.
(`SoftmaxInt16()` seems to follow a similar approach.)

### PR DESCRIPTION
**Pull request by AdithyaSrivastava01**
As a reference to ISSUE #88861, the essential changes are made to call exp_on_negative_values() only once. The exp is stored in the output_data(temp buffer), and later, the scaling operation is performed on the precomputed exponents.

Fixes #88861

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Comment by AdithyaSrivastava01** [comment]
Hey @mihaimaruseac @keerthanakadiri could you let me know whether the PR is ready to merge or should I make some changes?

**Comment by AdithyaSrivastava01** [comment]
Greetings @mihaimaruseac is it possible for you to share the potential issues that the code is facing?

**Comment by mihaimaruseac** [comment]
There seems to be numerical errors in the tests.

Can you run the tests locally? At least `tensorflow/lite/kernels/activations_test.cc`
